sdk/python/kfp/components/_component_store.py

```diff
@@ -0,0 +1,94 @@
+        <local-search-path>/<name>/component.yaml
+        <url-search-prefix>/<name>/component.yaml
+
+        If the digest is specified, then the search locations are:
+        <local-search-path>/<name>/versions/sha256/<digest>
+        <url-search-prefix>/<name>/versions/sha256/<digest>
+
+        If the tag is specified, then the search locations are:
+        <local-search-path>/<name>/versions/tags/<digest>
+        <url-search-prefix>/<name>/versions/tags/<digest>
+        
+        Args:
+            name:   Component name used to search and load the component artifact containing the component definition.
+                    Component name usually has the following form: group/subgroup/component
+            digest: Strict component version. SHA256 hash digest of the component artifact file. Can be used to load a specific component version so that the pipeline is reproducible.
+            tag:    Version tag. Can be used to load component version from a specific branch. The version of the component referenced by a tag can change in future.
+
+        Returns:
+            A factory function with a strongly-typed signature.
+            Once called with the required arguments, the factory constructs a pipeline task instance (ContainerOp).
+        '''
+        #This function should be called load_task_factory since it returns a factory function.
+        #The real load_component function should produce an object with component properties (e.g. name, description, inputs/outputs).
+        #TODO: Change this function to return component spec object but it should be callable to construct tasks.
+        if name is None:
+            raise TypeError()
+        if name.startswith('/') or name.endswith('/'):
+            raise ValueError('Component name should not start or end with slash: "{}"'.format(name))
+        
+        tried_locations = []
+
+        if digest is not None and tag is not None:
+            raise ValueError('Cannot specify both tag and digest')
+
+        if digest is not None:
+            path_suffix = name + '/' + self._digests_subpath + '/' + digest
+        elif tag is not None:
+            path_suffix = name + '/' + self._tags_subpath + '/' + tag
+            #TODO: Handle symlinks in GIT URLs
+        else:
+            path_suffix = name + '/' + self._component_file_name
+
+        #Trying local search paths
+        for local_search_path in self.local_search_paths:
+            try:
+                component_path = Path(local_search_path, path_suffix)
+                tried_locations.append(str(component_path))
+                if component_path.is_file():
+                    return comp.load_component_from_file(str(component_path))
+            except:
+                pass
+
+        #Trying URL prefixes
+        for url_search_prefix in self.url_search_prefixes:
+            try:
+                url = url_search_prefix + path_suffix
+                tried_locations.append(str(url))
```

Comment:
Do you need to empty tried_locations before you append anything? It may contain local paths.

Reply:
why? the local paths should have been tried already?

Reply:
I want the user to see all tried locations. Imaging the user has a local location with components, but he made some mistake in either component name or the local search path configuration. I want the user to see the failed local paths too, not just failed URLs.